### PR TITLE
Ft update registration endpoint

### PR DESCRIPTION
**Pull request by Daltimore**
#### What does this PR do?
* Updates Registration Endpoint
#### Description of Task to be completed?
* Update registration endpoint to send email confirmation mail to the user's email address
* The confirmation link should redirect the user back to the app to login after successful confirmation
* If email has been successfully confirmed, add the user to the DB
#### How should this be manually tested?
* Register as a new User 
#### Who should take a look?
@missating

### CODE REVIEW

**Comment by missating** [comment]
@Daltimore What differentiates a user that has verified their email, from one that has not? A user is supposed to be able to log in only when their email has been verified, going through your implementation I don't see that.

**Comment by Daltimore** [comment]
> @Daltimore What differentiates a user that has verified their email, from one that has not? A user is supposed to be able to log in only when their email has been verified, going through your implementation I don't see that.

The implementation checks if the user email already exists in the db, if it doesn't it will successfully register the user and then send a confirmation email then one of the listed todos was to send a confirmation link to the user to login which was added in the email template. the email verification here is checking if the email already exists before saving in the db

File: src/controllers/users.js [hunk 1]
```diff
@@ -23,11 +24,27 @@ export default {
...
     });
     const accessToken = generateToken(user.toJSON());
 
+    user.save((err) => {
+      if (err) {
+        return res.status(500).send({ msg: err.message });
+      }
+    }).then(() => {
+      confirmMail(user, res);
+    });
```
Comment (missating) [inline src/controllers/users.js]: @Daltimore This is only sending `user` and `res` no `url` where does it send the `url`?

Comment (Daltimore) [reply src/controllers/users.js]: I'm passing the URL in the template I will take it out from the helper function

**Pull request merged**